compiler/install

```diff
@@ -42,7 +42,7 @@ fpp-to-cpp
 fpp-to-xml
 "
 
-commit=`git log | awk 'NR == 1 { print $2 }'`
+commit=`git describe --tags --always`
```

Comment:
I think we need to use the full commit hash here, for development purposes. On the devel branch, we will need to check in dependencies to specific commits ahead of the last release tag. (We did this frequently while developing release 3.0.) Also, I am reluctant to introduce alternate/ambiguous ways to refer to the same commit. If we use the full has, there's no ambiguity (no danger of hash conflict, and no danger that a string comparison against the expected commit will fail in a surprising way).

Reply:
An alternative approach would be to replace the hardcoded version (e.g. v1.0.0) in the string with this version, and leave the commit hash so we have both.  Using this alternate, the released versions would look identical to what we have now...but the interim versions would include some extra information.

**Tag version Example**
```
fpp-depend v1.0.0 commit aabbccddeeff....
```

**Intermin Example**
```
fpp-depend v1.0.0-gaabbccdd commit aabbccddeeff....
```

This is the best of both worlds.  The exact hash is there but a human readable version+ is also available.  It also means we wouldn't have to remember to manually bump that version....

Reply:
...and it would give us a version in the same format that F´ uses to report versions.

Reply:
On further investigation, I think the full commit hash is not necessary. It looks like (1) `git describe` provides enough info to reconstruct the exact commit and (2) it does so in a form that other commands (e.g., `git checkout`) will accept.

Maybe we should replace the version stamp and the commit hash with the output of `git describe`. For example, at HEAD^ from v1.0.0, `git describe` says this:
```
[I] bocchino@MT-306179 ~/J/T/fpp ((23165a51…))> git describe --tags --always
v0.3.0-616-g23165a51
```
In this case maybe `fpp-depend --help` should say
```
fpp-depend v0.3.0-616-g23165a51
```
instead of what it says now
```
fpp-depend v1.0.0 commit 23165a5173848397743edffc675c94b0c57d58c2
```
We will also need to make the corresponding change to the version stamp in the docs. That is stored here:

https://github.com/fprime-community/fpp/blob/ee355fc99eb8040157c62e69f58ac6a8435cd981/defs.sh#L5

Since this is a nontrivial change to the version stamping mechanism of FPP, I'd like to move it out of this PR and into a separate issue or PR. Let's revert the change on this PR.

Reply:
I guess the 616 means "616 commits ahead of v0.3.0." I'm not sure what the `g` is doing in there ahead of the hash prefix. But ultimately it doesn't matter, if git knows unambiguously what it means.

Reply:
Sure, I can revert it out of the PR.  FYI the "g" means "git" to signify that those last 8 digits are a git commit hash in shortened form.  This was done such that the same tag format could be generated by some other source control using a unique identifier in their native format.